docs/docs/dynamo-store/architecture.md

```diff
@@ -0,0 +1,117 @@
+| `b`ytes     | ?Number                             | Only exists in the Tip, number of bytes in predecessor batches |
+| `etag`      | ?String                             | Etag of latest update (used for OCC)                           |
+| `e`vents    | List(Map([Event](#event-schema)))   | [Events](#event-schema)                                        |
+| `c`ases     | List(String)                        | Event types                                                    |
+| `u`nfolds   | List(Map([Unfold](#unfold-schema))) | [Unfolds](#unfold-schema)                                      |
+| `a`ppends   | Number                              | The number of events appended in the latest update             |
+| `n`umber    | Number                              | The version of the stream                                      |
+
+<h2 id="event-schema">Event schema</h2>
+
+| Key             | Type    | Description                                                  |
+| --------------- | ------- | ------------------------------------------------------------ |
+| `t`imestamp     | String  | ISO8601 DateTime representing the time the event was written |
+| `D`ata encoding | ?Number | the encoding used for the `d`ata attribute                   |
+| `d`ata          | ?Binary | The event payload                                            |
+| `M`eta encoding | ?Number | the encoding used for the `m`eta attribute                   |
+| `m`eta          | ?Binary | The event metadata                                           |
+| `x`actionId     | ?String | The correlation id of the event                              |
+| wh`y`           | ?String | The causation id (wh`y` did this event occur)                |
+
+<h2 id="unfold-schema">Unfold schema</h2>
+
+| Key             | Type    | Description                                                  |
+| --------------- | ------- | ------------------------------------------------------------ |
+| `i`ndex         | Number  | The index at which the unfold was generated                  |
+| `c`ase          | String  | The event type of the unfold                                 |
+| `t`imestamp     | String  | ISO8601 DateTime representing the time the event was written |
+| `D`ata encoding | ?Number | the encoding used for the `d`ata attribute                   |
+| `d`ata          | ?Binary | The event payload                                            |
+| `M`eta encoding | ?Number | the encoding used for the `m`eta attribute                   |
+| `m`eta          | ?Binary | The event metadata                                           |
+
+# Reactions
+
+Reactions are an often necessary part of event-sourced architectures. DynamoDB
+offers DynamoDB Streams, a Change Data Capture (CDC) for your table. This CDC
+feed offers two guarantees
+
+- Each stream record appears exactly once in the stream
```

Comment:
does 'stream record' relate to anything anyone will come here with some concept of?
maybe "every change (insert, update, delete) creates a single record in the DDB stream"

Reply:
terminology is ripped straight from https://docs.aws.amazon.com/amazondynamodb/latest/developerguide/Streams.html

Reply:
I'll add some more info before this about the feed

Reply:
yeah key thing is we have an overload on 'stream' that the rest of the world doesnt
and stream record means something if you're a DDB streams dude, but nothing if you're coming off the street